includes/class-experimental-abtest.php

```diff
@@ -88,6 +100,41 @@ public function get_variation( $test_name ) {
 		return $variation;
 	}
 
+
+	/**
+	 * Perform the request for a experiment assignment of a provided A/B test from WP.com.
+	 *
+	 * @param array $args Arguments to pass to the request for A/B test.
+	 * @return array|\WP_Error A/B test variation error on failure.
+	 */
+	public function request_assignment( $args ) {
```

Comment:
Could we make this `protected` like the previous `request_variation`? This is just to prevent external usage of the method and then us needing to maintain backwards compatibility of it into the future.

Reply:
> Could we make this protected like the previous request_variation? 

I think we need to keep this `public` because we call this method on the API side [here](https://github.com/woocommerce/woocommerce-admin/pull/8428/files#diff-c34adf1634556042d16d5c1e4d69b1c2205390c42c58bd0a433460cacf6ac6f4R78).